app/js/components/names.jsx

```diff
@@ -1,265 +1,241 @@
+    this.setState((prevState) => ({
+      display: 'view',
+      givenName: given === null ? '' : prevState.givenName,
+      middleName: middle === null ? '' : prevState.middleName,
+      familyName: family === null ? '' : prevState.familyName,
+      voided: prevState.response.preferredName.voided,
+      preferred: prevState.preferred,
+      creator: prevState.response.auditInfo.creator.display,
+      dateCreated: prevState.response.auditInfo.dateCreated,
+      errorGivenName: '',
+      errorFamilyName: ''
+    }));
+  };
+
+  render() {
+    let givenErrorClass = '';
+    let familyErrorClass = '';
+    if(this.state.errorGivenName && this.state.display === 'edit') {
+      givenErrorClass = 'has-error'
+    }
+    if(this.state.errorFamilyName && this.state.display === 'edit') {
+      familyErrorClass = 'has-error'
+    }
+    return(
+      <div>
+        <form className='form-horizontal'>
+          {this.state.display === 'view' &&
+          <div className='form-group'>
+              <div className='col-sm-2'>
+                  <input className='btn btn-success form-control'
+                         name='edit'
+                         type='button'
+                         value='Edit Name'
+                         onClick={this.handleEdit}/>
+              </div>
+          </div>}
+
+          <div className='form-group '>
+              <label className='control-label col-sm-2'>Preferred</label>
+              <div className='col-sm-2'>
+                  <input
+                      name='username'
+                      type='checkbox'
+                      checked={this.state.preferred}
+                      disabled={this.state.display === 'view' ?
+                       'disabled' : null}/>
+              </div>
+          </div>
+
+          <div className={'form-group ' + givenErrorClass}>
+              <label className='control-label col-sm-2'>
+              Given Name* </label>
+              <div className='col-sm-5'>
+                  <input  className='form-control'
+                          ref='givenname'
+                          name='givenName'
+                          type='text'
+                          value={this.state.givenName}
+                          onChange={this.handleChange}
+                          readOnly={this.state.display === 'view' ?
+                           'readonly' : null}
+                          required
+                           />
+              </div>
+              {this.state.errorGivenName &&
+              this.state.display === 'edit' &&
+              <div className='input'>{this.state.errorGivenName}</div>}
+          </div>
+
+          <div className='form-group '>
+              <label className='control-label col-sm-2'>
+              Middle Name  </label>
+              <div className='col-sm-5'>
+                  <input  className='form-control'
+                          ref='middlename'
```

Comment:
I notice you are using ref `uncontrolled` and onChange `controlled` form element together. I'm not really sure but I don't think that's a good idea. You might want to read up on react controlled and uncontrolled form element. You can check this [link](https://goshakkk.name/controlled-vs-uncontrolled-inputs-react/) as well

Reply:
Agreed. In my "rookieness" i thought i needed to use the "this.refs" to access element values, but as you pointed out earlier, my onChange method does a good job of updating my state values already. Thanks